### PR TITLE
chore: run eslint and prettier in the pipeline

### PR DESCRIPTION
**Pull request by ZhongpinWang**
## What Has Changed?
Closes SAP/cloud-sdk-backlog#723.

I modified the `package.json` to run lint and prettier together similar to the js repository. Then, a pipeline for checking lint and prettier is added so that formatting will not be forgotten in the future. Use `npm run lint:fix` locally to fix formatting automatically.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
~~- [ ] You formatted all changed files with prettier (`npm run prettier`)~~
- [x] You formatted all changed files with lint and prettier (`npm run lint:fix`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

**Review by jjtang1985 — APPROVED** [review]
LGTM.
I assume you've tested manually for the commands.

**Review by FrankEssenberger — APPROVED** [review]
LGTM

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Explain what you have changed.
   - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
   - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
 - [ ] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
-- [ ] You formatted all changed files with prettier (`npm run prettier`)
+- [ ] You formatted all changed files with lint and prettier (`npm run lint:fix`)
```
Comment (FrankEssenberger) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: [q] since the pipeline is red, when the linter fails is this checkbox needed?

Comment (ZhongpinWang) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: I favor removing all vale, lint, and build. These are all checked in the pipeline.

Comment (ZhongpinWang) [reply .github/PULL_REQUEST_TEMPLATE.md]: I updated the `PULL_REQUEST_TEMPLATE.md` and `CONTRIBUTING.md`.

**Pull request merged**